assets/src/data/model/entity-factory/create.js

```diff
@@ -0,0 +1,701 @@
+ * @param {Object} opts used to pass through additional options for the
+ * Object.defineProperty call.
+ */
+export const createGetter = ( instance, fieldName, fieldValue, opts = {} ) => {
+	Object.defineProperty( instance, fieldName, {
+		get() {
+			return fieldValue;
+		},
+		...opts,
+	} );
+};
+
+/**
+ * This creates a getter that calls the provided callback when invoked.
+ *
+ * The callback receives the `instance` argument passed through
+ *
+ * @param {Object} instance
+ * @param {string} propertyName
+ * @param {function(Object)} callBack
+ * @param {Object} opts
+ */
+export const createCallbackGetter = (
+	instance,
+	propertyName,
+	callBack,
+	opts = {}
+) => {
+	Object.defineProperty( instance, propertyName, {
+		get() {
+			return callBack( instance );
+		},
+		...opts,
+	} );
+};
+
+/**
+ * A generic getter and setter creator for a provided instance
+ *
+ * @param {Object} instance
+ * @param {string} fieldName
+ * @param {*}  initialFieldValue
+ * @param {Object} opts Optional, pass through options used by
+ * Object.defineProperty
+ */
+export const createGetterAndSetter = (
+	instance,
+	fieldName,
+	initialFieldValue,
+	opts = {}
+) => {
+	let propertyValue = initialFieldValue;
+	Object.defineProperty( instance, fieldName, {
+		get() {
+			return propertyValue;
+		},
+		set( receivedValue ) {
+			assertValidValueForPreparedField(
+				fieldName,
+				receivedValue,
+				instance
+			);
+			setSaveState( instance, SAVE_STATE.DIRTY );
+			propertyValue = receivedValue;
+		},
+		...opts,
+	} );
+};
+
+/**
+ * A getter and setter creator for an field alias.
+ *
+ * @param {Object} instance
+ * @param {string} originalFieldName
+ * @param {string} aliasFieldName
+ * @param {Object} opts
+ */
+export const createAliasGetterAndSetter = (
+	instance,
+	originalFieldName,
+	aliasFieldName,
+	opts = {},
+) => {
+	if ( originalFieldName !== aliasFieldName ) {
+		Object.defineProperty( instance, aliasFieldName, {
+			get() {
+				return instance[ originalFieldName ];
+			},
+			set( receivedValue ) {
+				return instance[ originalFieldName ] = receivedValue;
+			},
+			...opts,
+		} );
+	}
+};
+
+/**
+ * A getter creator for a field alias.
+ *
+ * @param {Object} instance
+ * @param {string} originalFieldName
+ * @param {string} aliasFieldName
+ * @param {Object} opts
+ */
+export const createAliasGetter = (
+	instance,
+	originalFieldName,
+	aliasFieldName,
+	opts = {},
+) => {
+	if ( originalFieldName !== aliasFieldName ) {
+		Object.defineProperty( instance, aliasFieldName, {
+			get() {
+				return instance[ originalFieldName ];
+			},
+			...opts,
+		} );
+	}
+};
+
+/**
+ * Creates a fluent setter on the provided instance for the given field name.
+ *
+ * @param {Object} instance
+ * @param {string} fieldName
+ * @param {Object} opts  Options for Object.defineProperty
+ */
+export const createFluentSetter = ( instance, fieldName, opts = {} ) => {
+	Object.defineProperty( instance, 'set' + upperFirst( fieldName ), {
+		get() {
```

Comment:
ya this one is a bit tricky :)  Fluent setters have to be set up this way because they are returning `this`.  Getters and Setters using `get propertyName` and `set propertyName` syntax are just handling what happens when you do:

```
MyObject.propertyName = 'something';
```
So for a fluent setter, I have the `get` set property actually return a `function` which allows the user to then call it.

Does that make sense?  If not, I can add further explanation.